Mage/src/main/java/mage/game/events/DamagedBatchCouldHaveFiredEvent.java

```diff
@@ -0,0 +1,15 @@
+package mage.game.events;
+
+/**
+ * Does not contain any info on damage events, and can fire even when all damage is prevented.
+ * Fire any time a DAMAGED_BATCH_FOR_ALL could have fired (combat & noncombat).
+ * It is not a batch event (doesn't contain sub events), the name is a little ambiguous.
+ *
+ * @author Susucr
+ */
+public class DamagedBatchCouldHaveFiredEvent extends GameEvent {
+
+    public DamagedBatchCouldHaveFiredEvent(boolean combat) {
+        super(EventType.DAMAGED_BATCH_COULD_HAVE_FIRED, null, null, null, 0, combat);
```

Comment:
Is there a reason for the "combat" flag in this event? (If so, include it in comment both here and in GameEvent)

Maybe "COULD_HAVE_DAMAGED" would be a clearer name (as it's neither a batch nor a damaged event), but not sure, current name is also okay.

Reply:
Not for the Phantom effects, but since all the damage related events do have that flag, I felt it was better to keep it there if we ever need it.

Reply:
As for the name, yeah I wasn't quite inspired with it.
It is a weird one and shouldn't be used directly in a trigger or replacement effect.